code/obj/item/kendo.dm

```diff
@@ -0,0 +1,250 @@
+                var/parry_block = S.parry_block_check(attacker,defender)
+                if((parry_block == 1) || (parry_block == 2))
+                    attacker.do_disorient((attacker.equipped().stamina_damage),0,0,0,0,1,null)
+                    return
+                else
+                    if((attacker.a_intent=="disarm") && prob(20))
+                        if(defender.equipped())
+                            var/obj/item/I = defender.equipped()
+                            defender.u_equip(I)
+                            I.set_loc(defender.loc)
+                            var/target_turf = get_offset_target_turf(I.loc,rand(5)-rand(5),rand(5)-rand(5))
+                            SPAWN_DBG(1 DECI SECOND)
+                                I.throw_at(target_turf,3,1)
+                                defender.show_text("<b>[attacker] knocks the [I] right out of your hands!</b>","red")
+                                attacker.show_text("<b>You knock the [I] right out of [defender]'s hands!</b>","green")
+        ..()
+
+    attack_hand(mob/user as mob)
+        if(src.loc != user)
+            change_guard(user,user.a_intent)
+        ..()
+
+    /*throw_begin(atom/target)
+        stat_reset()
+        ..()*/
+
+    dropped(mob/user as mob)
+        ..()
+        stat_reset()
+
+//==========
+//Shinai Bag
+//==========
+
+/obj/item/shinai_bag
+    name = "shinai bag"
+    desc = "竹刀袋 : a tube-like back for holding two shinai."
+    wear_image_icon = 'icons/mob/back.dmi'
+    icon_state = "shinaibag-closed"
+    item_state = "shinaibag-closed"
+    flags = ONBACK | FPRINT | TABLEPASS
+    w_class = 4.0
+    var/open = 0
+    var/shinai = 2
+
+    proc/update_spront(var/mob/user)
+        if(!open)
+            src.icon_state = "shinaibag-closed"
+            src.item_state = "shinaibag-closed"
+        else
+            src.icon_state = "shinaibag-[shinai]"
+            src.item_state = "shinaibag-[shinai]"
+
+        if(src.loc == user)
+            user.update_clothing()
+            user.update_inhands()
+
+    proc/draw_shinai(var/mob/user)
+        if(shinai)
+            var/obj/item/shinai/S = new /obj/item/shinai
+            user.put_in_hand_or_drop(S)
+            S.change_guard(user,user.a_intent)
+            shinai--
+            update_spront(user)
+
+        else
+            user.show_text("The [src] is empty!","red")
+
+    attack_self(mob/user as mob)
+        open = !open
+        update_spront(user)
+
+    attack_hand(mob/user as mob)
+        if(src.loc == user)
+            if(open)
+                draw_shinai(user)
+            else
+                open = !open
+                update_spront(user)
+        else if((user.a_intent == "grab") && open)
+            draw_shinai(user)
+        else
+            ..()
+
+    attackby(obj/item/W as obj, mob/user as mob)
+        if(istype(W,/obj/item/shinai) && open && (shinai<2))
+            user.u_equip(W)
+            qdel(W)
```

Comment:
Would prefer if this was added to the `contents` (and then that looked at first when drawing stuff from it), as that way if the item is changed in any way (e.g. someone uses Midas Touch on it to turn it to gold) you get the same item out when you draw from it again.

I'm fine with not instantiating shinai until they're first drawn from this, but once they're out it makes sense for them to be the same item from then on.

Reply:
I was thinking about doing something like that, but I didn't think having the server load of more item data would be worth it.

Reply:
Done in next commit :D